### PR TITLE
Enhancement/3376 implement a survey view trigger component

### PR DESCRIPTION
**Pull request by danielgent**
## Summary

Addresses issue #3376

## Relevant technical choices

**This is rebased off of Ivan's branch in CR, so may need updating & re-rebasing**

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/surveys/SurveyViewTrigger.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/**
```
Comment (eugene-manuilov) [inline assets/js/components/surveys/SurveyViewTrigger.js:1]: @danielgent please add the appropriate file header.

File: assets/js/components/surveys/SurveyViewTrigger.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	return null;
+};
+
+SurveyViewTrigger.propTypes = {
+	triggerID: PropTypes.string.isRequired,
+	ttl: PropTypes.number,
+};
+
+SurveyViewTrigger.defaultProps = {
+	ttl: null,
```
Comment (eugene-manuilov) [inline assets/js/components/surveys/SurveyViewTrigger.js:36]: The default value for the `ttl` should be `0`.

File: assets/js/components/surveys/SurveyViewTrigger.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+const { useSelect, useDispatch } = Data;
+
+const SurveyViewTrigger = ( { triggerID, ttl } ) => {
+	const usingProxy = useSelect( ( select ) => select( CORE_SITE ).isUsingProxy() );
+
+	const { triggerSurvey } = useDispatch( CORE_USER );
+
+	useMount( () => {
+		if ( usingProxy ) {
+			triggerSurvey( { triggerID, ttl } );
```
Comment (eugene-manuilov) [inline assets/js/components/surveys/SurveyViewTrigger.js:23]: The `triggerSurvey` action accepts two parameters: `triggerID` and `options`.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @danielgent! Found a few issues. Please, take a look.

File: assets/js/components/surveys/SurveyViewTrigger.js (hunk 2 above)
Comment (danielgent) [reply assets/js/components/surveys/SurveyViewTrigger.js:36]: So I don't read the AC or IB like that @eugene-manuilov  *BUT* I can totally see that from following other PRs

You may have an extra comment on how I do default values and destructuring though so I'm expecting this to be sent back again :smile:

File: assets/js/components/surveys/SurveyViewTrigger.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+const { useSelect, useDispatch } = Data;
+
+const SurveyViewTrigger = ( { triggerID, ttl } ) => {
+	const usingProxy = useSelect( ( select ) => select( CORE_SITE ).isUsingProxy() );
+
+	const { triggerSurvey } = useDispatch( CORE_USER );
+
+	useMount( () => {
+		if ( usingProxy ) {
+			triggerSurvey( { triggerID, options: { ttl } } );
```
Comment (eugene-manuilov) [inline assets/js/components/surveys/SurveyViewTrigger.js:41]: @danielgent this is still incorrect. Please, check the signature of the `triggerSurvey` action and update this call: https://github.com/google/site-kit-wp/blob/cfb5225d711d1eb6f812ca862f1494be91ba7566/assets/js/googlesitekit/datastore/user/surveys.js#L75-L85

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]

**Review by eugene-manuilov — APPROVED** [review]
Awesome! Thanks, @danielgent.

**Pull request merged**